src/System.CommandLine/Collections/AliasedSet.cs

```diff
@@ -69,34 +69,29 @@ private protected void EnsureAliasIndexIsCurrent()
                 return;
             }
 
-            var array = DirtyItems.ToArray();
-
-            for (var i = 0; i < array.Length; i++)
+            foreach (var dirtyItem in DirtyItems)
             {
-                var dirtyItem = array[i];
                 var aliases = GetAliases(dirtyItem).ToArray();
 
-                foreach (var pair in ItemsByAlias.Where(p => p.Value.Equals(dirtyItem)).ToArray())
+                foreach (var pair in ItemsByAlias)
                 {
                     if (pair.Value.Equals(dirtyItem))
                     {
                         ItemsByAlias.Remove(pair.Key);
                     }
                 }
 
-                var wasRemoved = !Items.Contains(dirtyItem);
-
-                if (!wasRemoved)
+                if (Items.Contains(dirtyItem))
```

Comment:
Are repeated items allowed?

I wonder if a `SortedList<T>` would perform better.

Reply:
I was considering `SortedList<T>` instead of `List<T>` if `HashSet<T>` is not an option.

`List<T>.Contains(T)` is O(N).

Reply:
Ah, got it. I'll take a look.